### DISCUSSION TITLE
What are you doing with Chirpity?

### DISCUSSION THREAD

**Discussion by Mattk70** [Mattk70/Chirpity-Electron#61] [Polls]
I'm curious to know how people use the results of the audio they analyse. This will help me understand better what the best ways might be to develop the "after analysis" features of the software.

Here's a quick poll. Also - if you have a story you'd like to share - use the show and tell to let me know about it!

**Comment by Vollpflock** [comment]
Initially, I went for the "Share online" option, but it turns out the csv file is all I need and I don’t know that else I want from it. I’m currently reporting observations by using a script, which is going smoothly (especially since there’s a column for latitude and longitude). 

A potential improvement could be a faster way to save detections—perhaps bypassing the record creation step with a keyboard shortcut. It would also be useful to have two categories for the observations: "confirmed" for the definite ones and "unconfirmed" for those that are uncertain and need further verification.

However, what I’m currently interested is in building my own sound library. 

I haven't found another convenient method for creating a library that includes both sound files and their spectrograms and think Chirpity can help a lot in these regards. So, I'm leaning more towards the archive feature now.

I'm not exactly sure what I'm looking for in it, though. I need to test first how the .wav files hold up after being compressed. Do you normalize them before compression, and which compression format are you using?

**Edit:**

Just as a thought: 
Maybe separating “Share online” and “Just keep them for my own use” more? Here's what I'm thinking: I go through the recordings and make new records for sounds I don't want to keep in the sound library, just for the sake of reporting them. In the end, my archive gets cluttered with a bunch of calls that I don't really want to keep but only created records for to report my observations.

I'd like to save the good sound clips for my library and the uncertain ones for later verification. Maybe we could have a shortcut for "reporting"? Like, adding an extra column in the CSV file labeled "Report" with "Yes" for the ones I'm reporting.

**Reply by Mattk70** [reply]
Thanks @Vollpflock, some interesting thoughts here.

> A potential improvement could be a faster way to save detections—perhaps bypassing the record creation step with a keyboard shortcut.

Not entirely sure what you're looking for here. Ctrl-S (or Command-S) will save detections. I think it might make sense that if you have a filtered view of just one species, you should be able to save the records for that species only.

>  I need to test first how the .wav files hold up after being compressed. Do you normalize them before compression, and which compression format are you using?

I have a script that converts the WAVs to M4A @24KHz sample rate, and I run it about once a month during the peak nocmig season. I don't normalise them, but my files' levels are ok to start with.

If I was building a sound library, I'd post the examples to a site like [Xeno-Canto](https://xeno-canto.org). I've been burned by HDD failure too many times in the past. The problem with these sites is they don't have an upload API, so you have to upload files using a form on their site. I think, one at a time. In the last couple of days I've been talking to one of the founders at XC about an upload API - so we'll see what comes of that. Ideally, you'd be able to select detections to upload to XC (or eBird/BirdTrack/Trektellen) directly from Chirpity, which could format the audio optimally for their site and pre-fill most of their required data fields. Fingers crossed!

Meanwhile, I'll have a think about more statuses for saved records. It's currently very basic, assuming that if you manually save a species / edit a record, it's 'confirmed'. There's an 'unknown sp.' option for uncertain IDs you want to save, but this can be a bit generic when there are many of them, and oftentimes you know its an either X or Y thing.

**Reply by Vollpflock** [reply]
Thank you!

> Not entirely sure what you're looking for here. Ctrl-S (or Command-S) will save detections. I think it might make sense that if you have a filtered view of just one species, you should be able to save the records for that species only.

I misspoke, I meant a quicker way to confirm detections.

> I have a script that converts the WAVs to M4A @24KHz sample rate, and I run it about once a month during the peak nocmig season. I don't normalise them, but my files' levels are ok to start with.

Interesting, I need to do that too. By the way, mine are already at 22kHz when recording. I need to do some testing to see if BirdNET performs the same as with files over 30kHz.

> If I was building a sound library, I'd post the examples to a site like [Xeno-Canto](https://xeno-canto.org). I've been burned by HDD failure too many times in the past. The problem with these sites is they don't have an upload API, so you have to upload files using a form on their site. I think, one at a time. In the last couple of days I've been talking to one of the founders at XC about an upload API - so we'll see what comes of that. Ideally, you'd be able to select detections to upload to XC (or eBird/BirdTrack/Trektellen) directly from Chirpity, which could format the audio optimally for their site and pre-fill most of their required data fields. Fingers crossed!

That's a very exciting idea. Automating that would be brilliant.

I generally find Xeno-Canto worth supporting, but it has always deterred me since you need to make a lot of entries when importing. I know you can make the workflow more efficient by sending files in batch, but it's still a pain. 

For integration into Chirpity (API), every mandatory field could be automized except "quality"?

Here's a list with all mandatory fields:
filename
country
location name
elevation
recording date
time of day
license
recordist name
type
automatic recording
genus
species
animal seen?
recording method
playback used?
quality


By the way, I haven't found an efficient way to compare spectrograms on xeno-canto efficiently. Do you happen to have an answer for that?

Every other image shows the waveform, and when you find a spectrogram, you can't click on it to listen in.
![Screenshot 2024-03-08 at 23 49 08](https://github.com/Mattk70/Chirpity-Electron/assets/26574621/99e6200b-1abf-48dd-8dd7-0854b63bd195)
![Screenshot 2024-03-08 at 23 49 23](https://github.com/Mattk70/Chirpity-Electron/assets/26574621/94aa454d-9ee1-4d2f-a2c4-7550d5a78c57)
![Screenshot 2024-03-08 at 23 49 41](https://github.com/Mattk70/Chirpity-Electron/assets/26574621/b1a8c2d8-3e7d-4bdc-a5fb-0fca46aba1a7)

> Meanwhile, I'll have a think about more statuses for saved records. It's currently very basic, assuming that if you manually save a species / edit a record, it's 'confirmed'. There's an 'unknown sp.' option for uncertain IDs you want to save, but this can be a bit generic when there are many of them, and oftentimes you know its an either X or Y thing.

Thanks, I've overlooked the 'unknown sp.' option so far."

**Reply by Mattk70** [reply]
XC does have an API for download. You can make a request for the spectrogram, audio, and metadata fields. It shouldn't be so hard to make use of this. So, instead of the "Compare species on XC" link opening a browser window, it **_could_** open a new Chirpity window. The window could have a carousel of the XC species' media files loaded with the Chirpity spectrogram view. You could organise the files by NFC, Flight Call, Call & Song - credit the contributor / include field notes beneath the spec. Hmm...

**Reply by Vollpflock** [reply]
Interesting idea, need to think about it

**Reply by Vollpflock** [reply]
Wow, you are already into it - awesome

**Reply by Mattk70** [reply]
It is looking (and working) pretty well. Sneak preview [here](https://chirpity.mattkirkland.co.uk/index2.html). Click the new image for a bigger view.

I'm working on eBird record format export, but I've run into issues with taxonomies. You'd think it'd all be joined up at Cornell, but apparently not...

**Reply by Vollpflock** [reply]
You never disappoint

**Comment by Vollpflock** [comment]
Jouuu and the comparison is even in the same color - awesome! Xeno-Canto was always a pain to navigate for me. 

Btw, have you already gotten a response from Xeno-Canto considering the Upload-API?

> If I was building a sound library, I'd post the examples to a site like [Xeno-Canto](https://xeno-canto.org/). I've been burned by HDD failure too many times in the past. The problem with these sites is they don't have an upload API, so you have to upload files using a form on their site. I think, one at a time. In the last couple of days I've been talking to one of the founders at XC about an upload API - so we'll see what comes of that. Ideally, you'd be able to select detections to upload to XC (or eBird/BirdTrack/Trektellen) directly from Chirpity, which could format the audio optimally for their site and pre-fill most of their required data fields. Fingers crossed!

**Reply by Mattk70** [reply]
Nothing from the guys at XC so far. I'll nudge them in a few days.

**Reply by Vollpflock** [reply]
Thought I'd check in again. Have you heard anything from XC in the meantime? Edit: Just saw it.

**Comment by David-Birdbrain** [comment]
I'd like to share my use case, not least to confirm I'm using the appropriate software :)
I run a re-wlding project in France. My goal is to leave my phone outside ~24hrs to record bird song. Do this 4-5 locations. Then import these recordings and get a record of which birds are present. I will store these in iNaturalist
Then repeat this each quarter.
My goal is to compare and contrast the project with land outside the project and to track changes in the project bird population over time 
Does Chiperty seem appropriate for this?
@Mattk70

**Reply by Mattk70** [reply]
Hi David, yes, I think Chirpity will be useful for this purpose. You will get candidate species IDs from the automated analysis, and need to verify these IDs. I'm not sure where in France your project is, but you may find some species you are interested in are missing from Chirpity's native model, since it is focused on species recorded in the UK. However, you can do your analysis using the BirdNET model too, if that is an issue. You can save your detections for each location and review changes over time, but it sounds like you'll be using iNaturalist for this aspect of your project.

**Reply by David-Birdbrain** [reply]
Thats great - Agree BirdNet was the model I was planning to use.
My main concern was whether the app could support multi- hour recordings

**Reply by richardhoeg** [reply]
I just discovered Chirpity. I am testing the combination of Chirpity in tandem with OwlSense, a birdsong listening device developed by the Owl Research Institute out of Montana. So far, so good! Chirpity ID'd four Great Horned Owl calls three nights ago, which BirdNET Analyzer did not find from the same exact wave files! I am a volunteer naturalist in a wilderness region named Sax-Zim Bog. Right now I have deployed the listening station way off the grid in the northern Boreal Forest. I will continue the test in about a week when I hike into the forest to retrieve files. My only question so far is there a way to add sound file excerpts with the birdsong record information to the archive? It would be nice not to have to save the entire sound file which has lots of "dead time" over the course of an entire night. Anyhow, thanks for your great work.

**Reply by Mattk70** [reply]
@richardhoeg - I'm glad to hear you are liking Chirpity! The way I manage "dead time" is to set the recorder to split the files every 5 minutes (or 1 minute, if you prefer). This has two practical benefits. Firstly, It looks like the AudioMoth and Song Meter recorders often corrupt the file they are recording to when their batteries die. If this is a long file, you'll lose all that recording, whereas if the files are just 1 or 5 minutes, you only lose that period of time.

With Chirpity, it's possible to save your audio to a library that's organised by recording location and date, in a folder location you choose (e.g. a NAS drive). The second benefit is that as a member you can choose to only save the files that have records in them to that library. So, if you're only interested in one species that calls infrequently, you only store the audio (1 or 5 minutes) with those calls in them.

<img width="427" height="313" alt="image" src="https://github.com/user-attachments/assets/3727b3fe-a006-4254-b59c-75a42761bdd5" />

**Comment by calyptura** [comment]
First of all, I want to thank you for developing Chirpity. It has been a game changer in helping me monitor bird communities in Brazil using autonomous recorders. I’d like to share how I’m using it, as it might make the tool even more valuable for me and other researchers.

I’ve developed a protocol where I listen to 48 minutes out of a total of 72 hours of monthly recordings from each autonomous recorder deployed in the field. I manually annotate all the species I detect in the 48 minutes I listen, which usually covers the most common birds in the area. Afterward, I run all 72 hours of recordings through Chirpity to expand the species list by identifying rarer and less common species.

Next, I run individual pattern-matching models, searching for the calls of all species — both those I identified manually and those Chirpity detected — across the entire dataset (72 hours). Chirpity has also been incredibly useful for facilitating the pattern-matching process by helping me quickly locate the best-quality recordings for each species within the dataset and use them as models for pattern matching.

I also tried using Raven Pro and BirdNET-Analyzer GUI for the same tasks, but Chirpity has proven to be much more useful for the protocol I developed, as well as being far more user-friendly.

Once again, thank you! 

Luciano

**Reply by Mattk70** [reply]
That's awesome, thanks for sharing!

**Comment by Giian07** [comment]
I record during the night using an old phone, then import the audio and analyze it for nocturnal flight calls! I've heard many stories of interesting birds calling and passing over my friends' houses during the night, so I wanted to monitor this migration myself. 
I've just started using the program and so far it's amazing. I've also tried BirdNET-Analyzer, but in my opinion Chirpity is more user-friendly, faster and doesn't crash as often. So I love it so far.
